osbs/build/build_request.py

```diff
@@ -183,6 +183,39 @@ def render_reactor_config(self):
             logger.debug("removing reactor_config plugin: no secret")
             self.dj.remove_plugin('prebuild_plugins', 'reactor_config')
 
+    def render_orchestrate_build(self):
+        phase = 'buildstep_plugins'
+        plugin = 'orchestrate_build'
+        if not self.dj.dock_json_has_plugin_conf(phase, plugin):
+            return
+
+        if self.spec.platforms.value is None:
+            logger.debug('removing {0} plugin: no platforms'.format(plugin))
+            self.dj.remove_plugin(phase, plugin)
+            return
+
+        if self.spec.client_config_secret.value is None:
+            logger.debug('removing {0} plugin: no secret'.format(plugin))
+            self.dj.remove_plugin(phase, plugin)
+            return
```

Comment:
It's a required param in the plugin. But that's a good point. It'd be more flexible if it gracefully defaults to whatever `Configuration` uses. I'll change the plugin to make it optional.